### PR TITLE
Feature/grapes tiptap editor

### PR DESCRIPTION
**Pull request by BrahimS**
@Zorin95670

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -22,6 +22,8 @@
 	"dependencies": {
 		"@intlify/vue-i18n-loader": "^3.3.0",
 		"@quasar/extras": "^1.0.0",
+		"@tiptap/core": "^2.0.0-beta.174",
```
Comment (Zorin95670) [inline frontend/package.json]: Don't use beta release, prefere to use the latest tiptap release 1.29.4

File: frontend/src/components/Builder/Grapes.vue [hunk 2]
```diff
@@ -5,30 +5,33 @@
 <script>
 import { onMounted } from "vue";
 import grapesjs from "grapesjs";
+import tipTap from "@tiptap/core";
+import tipTapStarter from "@tiptap/starter-kit";
 import "grapesjs/dist/css/grapes.min.css";
 import "grapesjs-preset-webpage/dist/grapesjs-preset-webpage.min.css";
 import "grapesjs-preset-webpage";
+console.log("tipTap: ", tipTap);
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove this console.log

File: frontend/src/components/Builder/Grapes.vue [hunk 3]
```diff
@@ -5,30 +5,33 @@
...
-				height: "800px",
 				width: "1100px",
 				storageManager: { autoload: 0 },
 				showOffsets: 1,
 				panels: { defaults: [] },
-				plugins: ["gjs-preset-webpage"],
+				plugins: ["gjs-preset-webpage", "@tiptap/core", "@tiptap/starter-kit"],
 				pluginsOpts: { "gjs-preset-webpage": {} },
 			});
+			console.log("editor: ", editor.editor.attributes.RichTextEditor);
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove this console.log

File: frontend/src/store/Users/mutations.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 export const GET_USERS = (state, theUsers) => (state.theUsers = theUsers);
 export const NEW_USER = (state, user) => state.theUsers.unshift(user);
+
 export const UPDATE_USER = (state, updatedUser) => {
-	const index = state.theUsers.findIndex((user) => user.id === updatedUser);
-	state.theUsers.splice(index, 1, updatedUser);
+	console.log("updatedUser: ", updatedUser);
```
Comment (Zorin95670) [inline frontend/src/store/Users/mutations.js]: Remove this console.log

File: frontend/src/store/Users/mutations.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 export const GET_USERS = (state, theUsers) => (state.theUsers = theUsers);
 export const NEW_USER = (state, user) => state.theUsers.unshift(user);
+
```
Comment (Zorin95670) [inline frontend/src/store/Users/mutations.js:3]: Move this file modification in another commit, because is not link to Install tipTap commit.

File: frontend/src/components/Builder/Grapes.vue [hunk 6]
```diff
@@ -5,12 +5,11 @@
 <script>
 import { onMounted } from "vue";
 import grapesjs from "grapesjs";
-import tipTap from "@tiptap/core";
+//import { Editor, EditorContent } from "@tiptap/core";
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove this comment

File: frontend/src/components/Builder/Grapes.vue [hunk 7]
```diff
@@ -5,12 +5,11 @@
...
 import tipTapStarter from "@tiptap/starter-kit";
 import "grapesjs/dist/css/grapes.min.css";
 import "grapesjs-preset-webpage/dist/grapesjs-preset-webpage.min.css";
 import "grapesjs-preset-webpage";
-console.log("tipTap: ", tipTap);
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: why did you remove now, instead of removing it in previous commit ?

File: frontend/src/components/Builder/Grapes.vue [hunk 8]
```diff
@@ -23,15 +22,14 @@ export default {
 				storageManager: { autoload: 0 },
 				showOffsets: 1,
 				panels: { defaults: [] },
-				plugins: ["gjs-preset-webpage", "@tiptap/core", "@tiptap/starter-kit"],
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Why do you add it in the first commit if you remove it ?

File: frontend/src/components/Builder/Grapes.vue [hunk 9]
```diff
@@ -23,15 +22,14 @@ export default {
...
+				plugins: ["gjs-preset-webpage"],
 				pluginsOpts: { "gjs-preset-webpage": {} },
 			});
-			console.log("editor: ", editor.editor.attributes.RichTextEditor);
 
 			return editor;
 		});
 
-		return { tipTap, tipTapStarter };
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Why do you add it in the first commit if you remove it ?

File: frontend/src/store/Auth/actions.js [hunk 10]
```diff
@@ -9,12 +9,9 @@ export const register = ({ commit }, credentials) => {
...
-			commit("SET_USER_DATA", data);
-			console.log("data: ", data);
-		})
-		.catch((err) => console.log(err));
+	return API.post("/user/login", credentials).then(({ data }) => {
+		commit("SET_USER_DATA", data);
+	});
 };
 
 export const currentUser = ({ commit }, user) => {
```
Comment (Zorin95670) [inline frontend/src/store/Auth/actions.js:17]: I post my comment here because I can't comment the update of gandalf submodule, but please move the modifiaction of gandlaf module in separate commit and not in the commit Fix log in issues -- Part Three -- + desactive tipTap editor from com…

File: frontend/src/components/Builder/Grapes.vue [hunk 11]
```diff
@@ -1,35 +1,48 @@
 <template>
-	<q-page id="gjs"></q-page>
+	<div>
+		<!-- <EditorContent :editor="rte" />s -->
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: remove this comment.

File: frontend/src/components/Builder/Grapes.vue [hunk 12]
```diff
@@ -1,35 +1,48 @@
...
+import { useEditor, EditorContent } from "@tiptap/vue-3";
+import StarterKit from "@tiptap/starter-kit";
+import CKEDITOR from "grapesjs-plugin-ckeditor";
 import "grapesjs/dist/css/grapes.min.css";
 import "grapesjs-preset-webpage/dist/grapesjs-preset-webpage.min.css";
 import "grapesjs-preset-webpage";
 
 export default {
 	name: "Grapes",
+	components: {
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove this.

File: frontend/src/components/Builder/Grapes.vue [hunk 13]
```diff
@@ -1,35 +1,48 @@
...
+		//EditorContent,
+	},
 	setup() {
+		const rte = useEditor({
+			content: "<p>I’m running Tiptap with Vue.js. 🎉</p>",
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: I think the text have to change

File: frontend/src/components/Builder/Grapes.vue [hunk 14]
```diff
@@ -1,35 +1,48 @@
...
 				storageManager: { autoload: 0 },
 				showOffsets: 1,
 				panels: { defaults: [] },
-				plugins: ["gjs-preset-webpage"],
+				plugins: ["gjs-preset-webpage", "grapesjs-plugin-ckeditor"],
 				pluginsOpts: { "gjs-preset-webpage": {} },
 			});
 
-			return editor;
+			console.log("editor: ", editor);
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: remove this console.log

File: frontend/src/components/Builder/Grapes.vue [hunk 15]
```diff
@@ -35,7 +35,7 @@ export default {
 				storageManager: { autoload: 0 },
 				showOffsets: 1,
 				panels: { defaults: [] },
-				plugins: ["gjs-preset-webpage", "grapesjs-plugin-ckeditor"],
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Why do you remove this, because you add it in previous commit ?

File: frontend/src/composables/WorkSpace/useDomain.js [hunk 16]
```diff
@@ -94,18 +94,17 @@ export default function useDomainData(props) {
...
 	};
 
-	const refreshDomain = async (id) => {
-		await store.dispatch("appDomain/fetchDomainById", id);
-		let data = computed(() => store.getters["appDomain/allDomaines"]);
-		currentDomainDataContent.value = await data.value;
-		domainTags.value = await data.value[0].Tags;
-		await getMenuData();
-	};
+	// const refreshDomain = async (id) => {
```
Comment (Zorin95670) [inline frontend/src/composables/WorkSpace/useDomain.js]: Remove comment instead of keeping it in the code.

File: frontend/src/pages/Home/PageHome.vue [hunk 17]
```diff
@@ -87,16 +87,16 @@
 
 <script>
 import { defineComponent, ref } from "vue";
-import PageContent from "../components/Content/PageContent";
-import FavoriteContent from "../components/Content/FavoriteContent";
-import AjaxBar from "../components/UI/Progress/AjaxBar";
-import Drawer from "../components/UI/Drawers/Drawer.vue";
-import { pageSizeTweak } from "../common/index";
-import AccountSettings from "components/UI/Profil/AccountSettings";
+import PageContent from "../../components/Content/PageContent";
```
Comment (Zorin95670) [inline frontend/src/pages/Home/PageHome.vue:90]: prefere to use "@" instead of "../../"

File: frontend/src/pages/Settings/PageSettings.vue [hunk 18]
```diff
@@ -61,12 +61,12 @@
 
 <script>
 import { ref } from "vue";
-import AjaxBar from "../components/UI/Progress/AjaxBar";
-import PageContent from "../components/Content/PageContent";
-import Drawer from "../components/UI/Drawers/Drawer.vue";
-import AccountSettings from "components/UI/Profil/AccountSettings";
-import SettingsNav from "../components/Navigations/SettingsNav";
-import { pageSizeTweak } from "../common/index";
+import AjaxBar from "../../components/UI/Progress/AjaxBar";
```
Comment (Zorin95670) [inline frontend/src/pages/Settings/PageSettings.vue]: prefere to use "@" instead of "../"

File: frontend/src/pages/Settings/PageSettingsEnviType.vue [hunk 19]
```diff
@@ -62,13 +62,13 @@
 
 <script>
 import { ref } from "vue";
-import AjaxBar from "../components/UI/Progress/AjaxBar";
-import PageContent from "../components/Content/PageContent";
-import Drawer from "../components/UI/Drawers/Drawer.vue";
+import AjaxBar from "components/UI/Progress/AjaxBar";
```
Comment (Zorin95670) [inline frontend/src/pages/Settings/PageSettingsEnviType.vue:65]: use relative path from @ instead of current.

File: frontend/src/pages/Auth/Login.vue [hunk 20]
```diff
@@ -13,8 +13,8 @@
 	</div>
 </template>
 <script>
-import LoginForm from "../components/UI/Form/LoginForm";
-import LoginNav from "../components/UI/Navigation/LoginNav";
+import LoginForm from "components/UI/Form/LoginForm";
```
Comment (Zorin95670) [inline frontend/src/pages/Auth/Login.vue:16]: Use relative path from @ instead of current path

File: frontend/src/components/Builder/Grapes.vue [hunk 21]
```diff
@@ -6,43 +6,32 @@
...
-				panels: { defaults: [] },
-				plugins: [webpage],
-				pluginsOpts: { "gjs-preset-webpage": {} },
-			});
-
-			console.log("editor: ", editor);
-		});
-
-		return { rte };
+		console.log("editor: ", editor);
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: remove this console.log

File: frontend/src/services/index.js [hunk 22]
```diff
@@ -8,7 +8,9 @@ if (localStorage.getItem("user")) {
 	API.interceptors.request.use(
 		(config) => {
 			let params = new URLSearchParams();
-			const { accessToken } = JSON.parse(localStorage.getItem("user"));
```
Comment (Zorin95670) [inline frontend/src/services/index.js:11]: Move this in another commit, because not link with the commit "Change rich text editor -- part One --"

File: frontend/src/services/index.js [hunk 23]
```diff
@@ -28,7 +30,9 @@ if (localStorage.getItem("user")) {
 	API.interceptors.response.use(
 		(config) => {
 			let params = new URLSearchParams();
-			const { accessToken } = JSON.parse(localStorage.getItem("user"));
```
Comment (Zorin95670) [inline frontend/src/services/index.js:31]: Move this in another commit, because not link with the commit "Change rich text editor -- part One --"

File: frontend/package.json [hunk 24]
```diff
@@ -28,6 +29,9 @@
 		"axios": "^0.21.2",
 		"browserfs": "^1.4.3",
 		"ckeditor": "^4.12.1",
+		"ckeditor4": "^4.18.0",
+		"ckeditor4-vue": "^2.0.0",
+		"ckeditor5": "^34.0.0",
```
Comment (Zorin95670) [inline frontend/package.json]: remove this unused dependencies. Or use only the 5 instead of 4

File: frontend/src/components/Builder/Grapes.vue [hunk 25]
```diff
@@ -1,37 +1,95 @@
...
+							{ name: "colors", items: ["TextColor", "BGColor"] },
+						],
+					},
+					"gjs-preset-webpage": {},
+				},
+			});
+
+			console.log("editor: ", editor);
+
+			// editor.setCustomRte({
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Well, no...
Please don't commit comment.

File: frontend/src/pages/Workspaces/PageWorkspace.vue [hunk 26]
```diff
@@ -16,7 +16,7 @@
 		</q-header>
 
 		<AjaxBar />
-		<Drawer v-model="drawer">
+		<!-- <Drawer v-model="drawer">
```
Comment (Zorin95670) [inline frontend/src/pages/Workspaces/PageWorkspace.vue]: Remove it, don't comment it

File: frontend/src/pages/Workspaces/PageWorkspace.vue [hunk 27]
```diff
@@ -43,7 +43,7 @@
 					<q-tree :nodes="menu" node-key="label" default-expand-all />
 				</div>
 			</template>
-		</Drawer>
```
Comment (Zorin95670) [inline frontend/src/pages/Workspaces/PageWorkspace.vue]: Remove it, don't comment it

File: frontend/src/pages/Workspaces/PageWorkspace.vue [hunk 28]
```diff
@@ -74,7 +74,7 @@ import { defineComponent, ref, computed } from "vue";
 import { useStore } from "vuex";
 import { useRouter } from "vue-router";
 import ModalStepper from "components/UI/Dialogs/ModalStepper.vue";
-import Drawer from "components/UI/Drawers/Drawer.vue";
```
Comment (Zorin95670) [inline frontend/src/pages/Workspaces/PageWorkspace.vue]: Remove it, don't comment it

File: frontend/src/pages/Workspaces/PageWorkspace.vue [hunk 29]
```diff
@@ -89,7 +89,7 @@ export default defineComponent({
 		ModalStepper,
 		CreationFormStepper,
 		AjaxBar,
-		Drawer,
```
Comment (Zorin95670) [inline frontend/src/pages/Workspaces/PageWorkspace.vue]: Remove it, don't comment it

File: frontend/package.json [hunk 30]
```diff
@@ -31,7 +31,6 @@
 		"ckeditor": "^4.12.1",
 		"ckeditor4": "^4.18.0",
 		"ckeditor4-vue": "^2.0.0",
-		"ckeditor5": "^34.0.0",
```
Comment (Zorin95670) [inline frontend/package.json]: well, if you don't use it, so don't add it in previous commit

File: frontend/src/components/Builder/Grapes.vue [hunk 31]
```diff
@@ -1,16 +1,23 @@
...
 import { onMounted } from "vue";
+import "grapesjs/dist/css/grapes.min.css";
 import grapesjs from "grapesjs";
 import "ckeditor4";
-import "ckeditor4-vue";
 import "grapesjs-plugin-ckeditor";
 import "grapesjs-preset-webpage";
-import "grapesjs/dist/css/grapes.min.css";
-import "grapesjs-preset-webpage/dist/grapesjs-preset-webpage.min.css";
+// import "grapesjs-preset-webpage/dist/grapesjs-preset-webpage.min.css";
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove it, don't comment it

File: frontend/src/components/Builder/Grapes.vue [hunk 32]
```diff
@@ -1,16 +1,23 @@
...
+		</div>
+	</div>
 </template>
 
 <script>
 import { onMounted } from "vue";
+import "grapesjs/dist/css/grapes.min.css";
 import grapesjs from "grapesjs";
 import "ckeditor4";
-import "ckeditor4-vue";
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: don't add it in previous comment

File: frontend/src/components/Builder/Grapes.vue [hunk 33]
```diff
@@ -22,13 +29,13 @@ export default {
...
 				width: "1100px",
-				storageManager: true,
+				storageManager: false,
 				showOffsets: 1,
 				panels: { defaults: [] },
-				plugins: ["gjs-plugin-ckeditor", "gjs-preset-webpage"],
+				plugins: ["gjs-plugin-ckeditor"],
 				pluginsOpts: {
 					"gjs-plugin-ckeditor": {
-						language: "en",
```
Comment (Zorin95670) [inline frontend/src/components/Builder/Grapes.vue]: Remove it, don't comment it

**Review by Zorin95670 — CHANGES_REQUESTED** [review]
I think you have to clean all your commit, because:
- content of some commit don't have any links with the commit name
- You add code in some commit then remove it in next commit, so please commit only usefull changes
- Please take care about don't keep useless comment in your code

File: frontend/src/components/Builder/Grapes.vue (hunk 2 above)
Comment (BrahimS) [reply frontend/src/components/Builder/Grapes.vue]: Ok, I will remove it because it is no longer used
